com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/app/discovery/ZigBeeNetworkDiscoverer.java

```diff
@@ -385,8 +389,14 @@ private boolean getAssociatedNodes(final int networkAddress) throws InterruptedE
             // Request extended response, start index for associated list is 0
             final IeeeAddressRequest ieeeAddressRequest = new IeeeAddressRequest(networkAddress, 1, startIndex);
             ieeeAddressRequest.setDestinationAddress(new ZigBeeEndpointAddress(networkAddress));
-            CommandResult response = networkManager.sendTransaction(ieeeAddressRequest, ieeeAddressRequest).get();
+            Future<CommandResult> resultFuture = networkManager.sendTransaction(ieeeAddressRequest, ieeeAddressRequest);
+            if (resultFuture == null) {
+                logger.debug("Transaction manager has been shutdown");
+                return false;
+            }
+            CommandResult response = resultFuture.get();
             if (response.isError()) {
+                logger.debug("Transaction command has not been successful");
```

Comment:
I haven't checked the logging elsewhere, but I feel that this log entry probably isn't needed as the failure should already be logged. Maybe I'm wrong, so if you think it's needed let's keep it, but I would be tempted in that case to make the log more useful and print the `response` as part of the log entry as it will say if it timed out or failed.

Reply:
Since the failure is already logged elsewhere, we can simply remove the logging here as well.